pkg/operator/controller/canary/daemonset_test.go

```diff
@@ -0,0 +1,60 @@
+package canary
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDesiredCanaryDaemonSet(t *testing.T) {
+	canaryImage := "openshift/hello-openshift:latest"
+
+	daemonset := desiredCanaryDaemonSet(canaryImage)
+
+	expectedDaemonSetName := types.NamespacedName{
+		Namespace: "openshift-ingress-canary",
+		Name:      "ingress-canary",
+	}
+
+	if !cmp.Equal(daemonset.Name, expectedDaemonSetName.Name) {
+		t.Errorf("Expected daemonset name to be %s, but got %s", expectedDaemonSetName.Name, daemonset.Name)
+	}
+
+	if !cmp.Equal(daemonset.Namespace, expectedDaemonSetName.Namespace) {
+		t.Errorf("Expected daemonset namespace to be %s, but got %s", expectedDaemonSetName.Namespace, daemonset.Namespace)
+	}
+
+	expectedLabels := map[string]string{
+		"ingress.openshift.io/canary": "canary_controller",
```

Comment:
If this label is used elsewhere, I suggest creating a const for the strings.

Reply:
Fixed. Thanks!